x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_add_edit/form/field_config.tsx

```diff
@@ -895,7 +895,27 @@ export const FIELD = (readOnly?: boolean): FieldMap => ({
       isEditFlow: isEdit,
     }),
     validation: () => ({
-      validate: (value) => Boolean(value.script),
+      validate: (value) => {
+        // return false if script contains import or require statement
+        if (
+          value.script?.includes('import ') ||
+          value.script?.includes('require(') ||
+          value.script?.includes('journey(')
+        ) {
+          return i18n.translate('xpack.synthetics.monitorConfig.monitorScript.invalid', {
+            defaultMessage:
+              'Monitor script is invalid. Inline scripts cannot be full journey scripts. It must be only steps definitions.',
```

Comment:
```suggestion
              'Monitor script is invalid. Inline scripts cannot be full journey scripts, they may only contain step definitions.',
```

Reply:
I think even with this explanation it could be confusing what's wrong. It could be nice to link to documentation explaining we don't allow `import` or `require` or `journey` keywords in scripts, or it'd be cool if we could highlight the offending line in the editor.